utils/src/utils/apiUtils.ts

```diff
@@ -1,5 +1,30 @@
-import * as vscode from 'vscode';
-import type { Extension } from 'vscode';
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+import { Extension } from "vscode";
+import { AzureExtensionApi, AzureExtensionApiProvider } from "../../api";
+import { getPackageInfo } from "../getPackageInfo";
+
+class ExtensionNotFoundError extends Error {
+    constructor(extensionId: string) {
+        super(`Extension with id ${extensionId} not found.`);
+    }
+}
+
+export async function getAzureExtensionApi<T extends AzureExtensionApi>(extensionId: string, apiVersionRange: string): Promise<T> {
```

Comment:
If there are other things added to the `GetApiOptions` aside from extension id, it's entirely possible the caller would need to be able to specify them. I think we should go ahead and accept it, setting `extensionId` by default if it isn't present.

Reply:
Good idea